class.jetpack.php

```diff
@@ -487,6 +487,8 @@ private function __construct() {
 			Jetpack_Heartbeat::init();
 		}
 
+		add_filter( 'rest_authentication_errors', array( $this, 'wp_rest_authenticate' ) );
```

Comment:
Good catch - I think you're right.  This would cause auth here to override the user set by a cookie, for example.  `WP_REST_OAuth1` [only uses `rest_authentication_errors`](https://github.com/WP-API/OAuth1/blob/339fa7b38ce06d1ee98ca4dcb99ce45b64118322/lib/class-wp-rest-oauth1.php#L213-L225) to report errors that were stored previously.